cmd/codegen/generator/typescript/templates/src/method_solve.ts.gtpl

```diff
@@ -77,8 +77,9 @@
 
       		{{- with $optionals }}
       			{{- if $required }}, {{ end }}
-				{{- "" }}...opts{{- if gt (len $enums) 0 -}}, __metadata: metadata{{- end -}}
+				{{- "" }}...opts
 			{{- end }}
+      {{- if gt (len $enums) 0 -}}, __metadata: metadata{{- end -}}
```

Comment:
cc @TomChv - I *think* this is right.

Previously, we never had any required enum arguments. When this patch added them, then the defined `metadata` would never be attached (which caused a linting/codegen failure). I think this is the right fix, but just wanted to make sure.

Side note: this file is *weirdly* formatted, and uses an inconsistent mixture of tabs and spaces, which makes it fairly challenging to view/edit. Didn't want to fix this right in this PR to avoid a massive diff, but we should fix this.

Reply:
> Side note: this file is weirdly formatted, and uses an inconsistent mixture of tabs and spaces, which makes it fairly challenging to view/edit. Didn't want to fix this right in this PR to avoid a massive diff, but we should fix this.

Yeah this part needs a refactor but is not critical enough to be a priority